sky/serve/service_spec.py

```diff
@@ -60,13 +69,41 @@ def __init__(
                     'Currently, SkyServe will cleanup failed replicas'
                     'and auto restart it to keep the service running.')
 
+        if spot_policy is not None:
+            if autoscaler is not None or spot_placer is not None:
+                with ux_utils.print_exception_no_traceback():
+                    raise ValueError(
+                        'Cannot specify `spot_policy`, `autoscaler` and '
+                        '`spot_placer` at the same time.')
+            # TODO(MaoZiming): do not hardcode the name
+            if spot_policy == 'SpotHedge':
+                autoscaler = 'SpotOnDemandRequestRateAutoscaler'
+                spot_placer = 'DynamicFailover'
+            else:
+                with ux_utils.print_exception_no_traceback():
+                    raise ValueError(f'Unsupported spot policy: {spot_policy}.')
+
+        if autoscaler not in autoscalers.Autoscaler.get_autoscaler_names():
```

Comment:
What if `autoscaler is None`?

Reply:
it will raise an error in 
```
self._autoscaler: autoscalers.Autoscaler = (
            autoscalers.Autoscaler.from_spec(service_spec))
```
Maybe we can set a default autoscaler?

Reply:
Yeah we should have a default autoscaler. Maybe OnDemand request rate?